layout/layout.go

```diff
@@ -0,0 +1,740 @@
+	layers, err := i.Image.Layers()
+	if err != nil {
+		return nil, err
+	}
+
+	layer, err := findLayerWithSha(layers, sha)
+	if err != nil {
+		return nil, err
+	}
+
+	return layer.Uncompressed()
+}
+
+// AddLayer adds an uncompressed tarred layer to the image
+func (i *Image) AddLayer(path string) error {
+	layer, err := tarball.LayerFromFile(path)
+	if err != nil {
+		return err
+	}
+	image, err := mutate.AppendLayers(i.Image, layer)
+	if err != nil {
+		return errors.Wrap(err, "add layer")
+	}
+	i.mutateImage(image)
+	return nil
+}
+
+func (i *Image) AddLayerWithDiffID(path, diffID string) error {
+	// this is equivalent to AddLayer in the layout case
+	// it exists to provide optimize performance for local images
+	return i.AddLayer(path)
+}
+
+func (i *Image) ReuseLayer(sha string) error {
+	layer, err := findLayerWithSha(i.prevLayers, sha)
+	if err != nil {
+		return err
+	}
+	image, err := mutate.AppendLayers(i.Image, layer)
+	if err != nil {
+		return err
+	}
+	i.mutateImage(image)
+	return nil
+}
+
+func findLayerWithSha(layers []v1.Layer, diffID string) (v1.Layer, error) {
+	for _, layer := range layers {
+		dID, err := layer.DiffID()
+		if err != nil {
+			return nil, errors.Wrap(err, "get diff ID for previous image layer")
+		}
+		if diffID == dID.String() {
+			return layer, nil
+		}
+	}
+	return nil, fmt.Errorf("previous image did not have layer with diff id %q", diffID)
+}
+
+func (i *Image) Save(additionalNames ...string) error {
+	if len(additionalNames) > 1 {
+		return errors.Errorf("multiple additional names %v are not allow when OCI layout is used", additionalNames)
+	}
+
+	err := i.mutateCreatedAt(i.Image, v1.Time{Time: i.createdAt})
+	if err != nil {
+		return errors.Wrap(err, "set creation time")
+	}
+
+	cfg, err := i.Image.ConfigFile()
+	if err != nil {
+		return errors.Wrap(err, "get image config")
+	}
+	cfg = cfg.DeepCopy()
+
+	layers, err := i.Image.Layers()
+	if err != nil {
+		return errors.Wrap(err, "get image layers")
+	}
+	cfg.History = make([]v1.History, len(layers))
+	for j := range cfg.History {
+		cfg.History[j] = v1.History{
+			Created: v1.Time{Time: i.createdAt},
+		}
+	}
+
+	cfg.DockerVersion = ""
+	cfg.Container = ""
+	err = i.mutateConfigFile(i.Image, cfg)
+	if err != nil {
+		return errors.Wrap(err, "zeroing history")
+	}
+
+	// initialize path
+	path, err := Write(i.path, empty.Index)
+	if err != nil {
+		return err
+	}
+
+	annotations := map[string]string{}
+	if len(additionalNames) == 1 {
+		annotations["org.opencontainers.image.ref.name"] = additionalNames[0]
+	}
+
+	var diagnostics []imgutil.SaveDiagnostic
+	err = path.AppendImage(i.Image, WithAnnotations(annotations))
+	if err != nil {
+		diagnostics = append(diagnostics, imgutil.SaveDiagnostic{ImageName: i.Name(), Cause: err})
+	}
+
+	if len(diagnostics) > 0 {
+		return imgutil.SaveError{Errors: diagnostics}
+	}
+
+	return nil
+}
+
+func (i *Image) SaveFile() (string, error) {
+	return "", errors.New("not yet implemented")
+}
```

Comment:
What are the differences in output here? Between a local image `SaveFile()` and a layout image `Save()`?

Reply:
That sounds about right to me